dist/releases/embeds.go

```diff
@@ -40,3 +47,43 @@ func GetConfigFor(version string) ([]byte, error) {
 
 	return nil, fmt.Errorf("embedded config for version %s is not available; use one of %+v", version, GetVersions())
 }
+
+// GetJavaFile returns the java source file.
+func GetJavaFile() (fileFullPath, fileName string, err error) {
+	bytes, err := configs.ReadFile("java/" + JavaFips)
+	if err != nil {
+		return "", "", err
+	}
+
+	return createTempFile(bytes, "fipsJava")
+}
+
+// GetAlgorithmFile returns a file with disabled algorithms to check.
+func GetAlgorithmFile(javaDisabledAlgorithms []string) (fileFullPath, fileName string, err error) {
+	r := bytes.NewBuffer([]byte{})
+	for i := range javaDisabledAlgorithms {
+		r.WriteString(javaDisabledAlgorithms[i] + "\n")
+	}
+
+	return createTempFile(r.Bytes(), "disabledAlgorithms")
+}
+
+// createTempFile returns a temporary file with specified content and prefix, created in the OS's default temp folder
+// Closing of the file (defer os.Remove(file.Name())) is the caller's responsibility
+func createTempFile(content []byte, prefix string) (fileFullPath, fileName string, err error) {
+	file, err := os.CreateTemp(os.TempDir(), prefix)
+	if err != nil {
+		return "", "", err
+	}
+	fileModeRO := os.FileMode(syscall.S_IRUSR) | os.FileMode(syscall.S_IRGRP) | os.FileMode(syscall.S_IROTH)
```

Comment:
If any of the if statements here error the temp file isn't cleaned up. This function should defer cleanup the tempfile.